api/layer/container.go

```diff
@@ -84,6 +84,11 @@ func (n *layer) containerList(ctx context.Context) ([]*data.BucketInfo, error) {
 		res []cid.ID
 		rid = api.GetRequestID(ctx)
 	)
+
+	if api.IsAnonymousRequest(ctx) {
```

Comment:
Why only this request? I think we can have the same in other ones like object get.

Reply:
Moved this check to the old place, inside `Owner` func. After that it will work like previous implementation

Reply:
But what about keys? I suppose we should use a different key when we're doing anonymous requests. Maybe `layer`  should return `user.Signer` now to provide both at the same time (gw normally and anon if no auth)?

Reply:
I revised the code one more time.
We have the next component chain (gate component names):
* layer -> neofs -> pool

`layer` - handles API requests, calls `neofs` component to prepare data and then it call `pool` to execute request in neofs node.

In fact, the layer even doesn't need to know something about a key or signer. We will drop the key from it.
The _neofs_ component stores the same key (pool also stores it) and it is a good place to make magic with anonymous keys. We just need to pass isAnonymous flag to _neofs_ component and anonymous signer will be used

Reply:
Made changes in separate commits. At first look, I replaced all places when we cat override the signer. In other places, random keys are already in use by SDK client.

If these changes ok, even after few iterations, I will squash them to proper commit, if required